Add information about Distinguished Names (#2828)
theforeman/foreman-documentation: guides/common/modules/proc_creating-a-custom-ssl-certificate.adoc

[ alt_names ]
DNS.1 = _{context}.example.com_
----

> Optional: If you want to add Distinguished Name (DN) details to the CSR, add the following information to the `[ req_distinguished_name ]` section:
> +
> [options="nowrap", subs="+quotes,attributes"]
> ----
> [req_distinguished_name]
> CN = _{context}.example.com_
> countryName = Country Name (2 letter code)
> stateOrProvinceName = State or Province Name (full name)
> localityName = Locality Name (eg. city)
> organizationName = Name of the Organization or Company
> organizationalUnitName = Organizational Unit Name (eg. section)
> ----

Reviewer: I recommend using callouts.
```suggestion
. Optional: If you want to add Distinguished Name (DN) details to the CSR, add the following information to the `[ req_distinguished_name ]` section:
+
[options="nowrap", subs="+quotes,attributes"]
----
[req_distinguished_name]
CN = _{context}.example.com_
countryName = _My_Country_Name_ <1>
stateOrProvinceName = _My_State_Or_Province_Name_ <2>
localityName = _My_Locality_Name_ <3>
organizationName = _My_Organization_Or_Company_Name_
organizationalUnitName = _My_Organizational_Unit_Name_ <4>
----
<1> Two letter code.
<2> Full name.
<3> For example, a city name.
<4> For example, a section.
```

[ alt_names ]
DNS.1 = _{context}.example.com_
----
Optional: If you want to add Distinguished Name (DN) details to the CSR, add the following information to the `[ req_distinguished_name ]` section:
+
[options="nowrap", subs="+quotes,attributes"]
----
[req_distinguished_name]
CN = _{context}.example.com_
countryName =_My_Country_Name_ <1>
stateOrProvinceName = _My_State_Or_Province_Name_ <2>
localityName = _My_Locality_Name_ <3>
organizationName = _My_Organization_Or_Company_Name_
organizationalUnitName = _My_Organizational_Unit_Name_ <4>
----
<1> Country Name (2 letter code)
<2> State or Province Name (full name)

> <3> Locality Name (eg. city)

Reviewer 2: Latin abbreviations can cause problems for localization.

```suggestion
<3> Locality Name (example: city)
```

> <1> Two letter code
> <2> Full name
> <3> Full name (example: New York)
> <4> Division responsible for the certificate (example: IT)

Reviewer: Going by the format we use in this module (line 88), callouts should be ended with a dot. Also, I strongly recommend using **IT department** instead of **IT**, which just feels incomplete.

Ack @apinnick?
```suggestion
<1> Two letter code.
<2> Full name.
<3> Full name (example: New York).
<4> Division responsible for the certificate (example: IT department).
```

Reviewer 3: +1 for "IT department" because otherwise, it looks like a two letter code. :it:

> <1> Two letter code
> <2> Full name
> <3> Full name (example: New York)
> <4> Division responsible for the certificate (example: IT department)

Reviewer: ```suggestion
<1> Two letter code.
<2> Full name.
<3> Full name (example: New York).
<4> Division responsible for the certificate (example: IT department).
```

Author: Removed periods from the other callbacks.

> <4> Organizational Unit Name (eg. section)

Reviewer 2: ```suggestion
<4> Organizational Unit Name (example: IT department)
```
or anything else that makes sense.

Author: Changed this to IT.

> <4> Organizational Unit Name (example: IT)

Reviewer: I think the stuff you have here outside of parentheses is redundant, because it is already in the code block. What do you think @apinnick?

Reviewer 2: Not sure. If there are constraints (2-letter country code, for example), we need to call them out. If the other fields are self-evident, then they probably do not need callouts. I imagine that state/province would not need to be documented. 

My only question is whether all these fields are required. Would it be possible to add a DN without a state/province, for example? If a user lives in Vatican City, that would be relevant. OTOH, if the fields are not validated in any way, then we would not need detailed documentation.

Reviewer: Constraints are perfectly fine. What I meant was not to repeat the actual name of the field like this:
```suggestion
<1> 2 letter code.
<2> Full name.
<3> For example, New York
<4> For example, IT department
```

Author: @apinnick some of these fields are optional, however, we have a bug asking that this information be expressly mentioned in the CSR section to increase trust.

Author: @apinnick I have added "Optional" at the beginning of the preceding sentence. Does that work?

Reviewer 2: LGTM
